pkg/sideinputs/initializer/initializer.go

```diff
@@ -132,9 +132,6 @@ func startSideInputInitializer(ctx context.Context, store kvs.KVStorer, mountPat
 // gotAllSideInputVals checks if values for all side-inputs
 // have been received from the KV bucket
 func gotAllSideInputVals(sideInputs []string, m map[string][]byte) bool {
-	if len(sideInputs) != len(m) {
```

Comment:
`len(m)` is total number of side inputs defined in the pipeline, `len(sideInputs)` is the number of side inputs current vertex is watching.